src/main/java/org/testeditor/testenv/vagrant/VagrantTestExecutionMojo.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ *******************************************************************************/
+package org.testeditor.testenv.vagrant;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+@Mojo(name = "testExec")
+public class VagrantTestExecutionMojo extends AbstractVagrantMojo {
+
+	@Parameter(property = "project.build.directory")
+	private File outputPath;
+
+	@Parameter(property = "projectDirInTestEnv")
+	private String projectDirInTestEnv;
+
+	@Parameter(property = "mavenReporInTestEnv")
+	private String mavenReporInTestEnv;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		String command;
+		try {
+			String test = "";
+			if (System.getProperties().containsKey("test")) {
+				test = System.getProperty("test");
+			}
+			command = createExecCommand(test);
+		} catch (IOException e) {
+			throw new MojoExecutionException("", e);
+		}
+		int exitCode = executeVagrantCommand("vagrant", "ssh", "-c", command);
+		if (exitCode > 0) {
+			throw new MojoFailureException("Error executing tests. " + command);
+		}
+	}
+
+	/**
+	 * Creates an script to execute the tests in the target system and returns
+	 * the command to launch it with vagrant.
+	 * 
+	 * @param testStructure
+	 *            to be executed as test.
+	 * @return string with the command used in the vagrant ssh call.
+	 * @throws IOException
+	 *             on failure.
+	 */
+	protected String createExecCommand(String testName) throws IOException {
+		List<String> lines = Files.readAllLines(new File(getVagrantFileDir(), "Vagrantfile").toPath(),
+				StandardCharsets.UTF_8);
+		boolean isLinux = true;
+		for (String string : lines) {
+			if (!string.trim().startsWith("#")) {
+				if (string.contains("config.vm.communicator") && string.contains("winrm")) {
+					isLinux = false;
+				}
+			}
+		}
+		String executionScript = null;
+		String execCommand = null;
+		String execScript = null;
+		if (isLinux) {
+			executionScript = getExecutionScriptForLinux(testName);
```

Comment:
Yes thats right. At the current state we have no contract for vagrant images. The script for Linux tries a faster approach to get a clean environment by restarting the x session.
The windows script is used to avoid the provisioning step at development time. It is a shorter turn around cycle to change the execution script on the same vagrant image instead of using provision.
I think we should rework this after some more experiences with a vagrant based solution to define a image contract.